org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacProblem.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+* Copyright (c) 2024 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.core.dom;
```

Comment:
Could this be moved to the internal.javac package so it doesn't leak too much in existing JDT packages if we can contribute it upstream later?

Reply:
I will leave the JavacProblem in the dom package, because the downstream language server needs to access the JavacProblem object to get the Javac Code.

Reply:
OK, so should we create a public `org.eclipse.jdt.core.javac` package to put such public classes in while still having them separated from the upstream package?

Reply:
Moved it to internal. We can use the attributes field of existing CategorizedProblem class to store additional JavacCode information, no need to access the JavacProblem directly.